flake.nix

```diff
@@ -7,64 +7,95 @@
 
   outputs = { self, nixpkgs }: {
 
-    nixosModules.garnix = { lib, config, ... }:
+    nixosModules.default = { lib, config, ... }:
       let
-        cfg = config.garnix.persistence;
+        cfg = config.garnix.server;
       in
       {
+        options.garnix = {
+          server = {
```

Comment:
I kind of liked the idea of leaving this option on level up, i.e. `garnix.enable`. And then we'd have `garnix.persistence.enable` in addition. Or would that be too confusing? Not sure how idiomatic that is.

Reply:
It just feels not clear/explicit enough. It's potentially enough in the context of a `nixosConfiguration` module.

I'm also worried about future features that might make this even more confusing (which feature are we enabling?).

Reply:
Yeah, fair enough.